gulpfile.babel.js

```diff
@@ -57,29 +63,58 @@ function lintGulpfile() {
   return lint('gulpfile.babel.js');
 }
 
-function build() {
-  return gulp.src(path.join('src', config.entryFileName + '.js'))
-    .pipe($.plumber())
-    .pipe(webpackStream({
-      output: {
-        filename: exportFileName + '.js',
-        libraryTarget: 'umd',
-        library: config.mainVarName
-      },
-      module: {
-        loaders: [
-          { test: /\.js$/, exclude: /node_modules/, loader: 'babel-loader' }
-        ]
-      },
-      devtool: 'source-map'
-    }))
-    .pipe(gulp.dest(destinationFolder))
-    .pipe($.filter(['*', '!**/*.js.map']))
-    .pipe($.rename(exportFileName + '.min.js'))
-    .pipe($.sourcemaps.init({ loadMaps: true }))
-    .pipe($.uglify())
-    .pipe($.sourcemaps.write('./'))
-    .pipe(gulp.dest(destinationFolder));
+function build(done) {
+  rollup.rollup({
+    entry: path.join('src', config.entryFileName + '.js'),
+    plugins: [
+      nodeResolve({
+        jsnext: true,
+        /*
+         * Uncomment the following line to specify external
+         * deps to exclude from the bundle. By default, everything
+         * is bundled into the build
+         */
+        // skip: ['jquery'],
+        main: true
+      }),
+      rollupBabelPlugin({
+        sourceMaps: true,
+        presets: [rollupBabelEs2015Present],
+        babelrc: false
+      }),
+      // This allows you to require in CJS modules
+      commonjsPlugin(),
+      // This allows you to require in JSON files
+      jsonPlugin()
+    ]
+  }).then(function(bundle) {
+    var result = bundle.generate({
+      format: 'umd',
+      sourceMap: 'inline',
+      sourceMapSource: config.entryFileName + '.js',
+      sourceMapFile: exportFileName + '.js',
+      moduleName: config.mainVarName
+    });
+    var code = `${result.code}\n//# sourceMappingURL=./${exportFileName}.js.map`;
+
+    // Write the generated sourcemap
+    mkdirp.sync(destinationFolder);
```

Comment:
dunno. I basically copy+pasted it from some work @megawac did and haven't spent much time analyzing/refactoring it yet. I'll definitely check that other implementation out :v: